docs/overview/deep_dive/building_the_docs_pipline.rst

```diff
@@ -0,0 +1,482 @@
+Building the Docs Pipeline
+==========================
+
+.. _Sphinx: http://sphinx-doc.org/
+.. _Sphinx configuration file: https://www.sphinx-doc.org/en/master/usage/configuration.html
+.. _autosummary: https://www.sphinx-doc.org/en/master/usage/extensions/autosummary.html
+
+To build our docs, we use `Sphinx`_. Sphinx is an extendable documentation generator
+for Python. As our building pipeline is complex, we heavily customize Sphinx using 
+custom and third party extensions. As well as having a convenience script to build
+the docs.
+
+How the doc-builder is being run
+--------------------------------
+
+There are 2 ways to build the docs:
+
+1. Through a convenience script, which is useful for local development.
+2. Through a Docker image, which is the recommended way.
+
+We will go through how they work in the following sections.
+
+The convenience script
+~~~~~~~~~~~~~~~~~~~~~~
+
+``make_docs_without_docker.sh`` is a convenience script to build the docs. It takes 
+one argument, the path to a project to document. The project should have the following
+characteristics:
+
+1. It should have a ``requirements.txt``, or alternatively a ``requirements`` folder,
+   which includes a ``requirements.txt`` and an optional ``optional.txt`` file.
+
+2. It can have an optional ``optional.txt`` file, if not the script the script will
+   simply ignore it.
+
+3. It should have a ``docs`` folder, which contains an ``index.rst`` file. This file
+   is the root of the documentation.
+
+4. It can contain an optional ``docs/prebuild.sh`` file, which will be executed before
+   the docs are built. This is useful if you need to install some dependencies for the
+   docs to build.
+
+5. It can contain an optional ``docs/partial_conf.py`` which is a partial `Sphinx
+   configuration file`_.
+   This file will be imported with the default ``conf.py`` file located in the 
+   ``doc-builder`` repo.
+
+Running the script like this:
+
+.. code-block:: bash
+
+    ./make_docs_without_docker.sh /path/to/project
+
+will build the docs for the project and output them in the ``docs/build`` folder.
```

Comment:
will result in the creation of documentation for the project in the directory ``doc/build``